### PR TITLE
Created tracking on focus

### PR DESCRIPTION
**Pull request by AlKulinski**

### CODE REVIEW

File: src/app/admin-api/admin-api.component.ts [hunk 1]
```diff
@@ -46,6 +47,8 @@ export class AdminAPIComponent implements OnInit {
   ngOnInit (): void {
     const id = this.route.snapshot.paramMap.get('id')
     const person = this.data.find(p => p.id === Number(id))
+    const elementToSelect = this._elementRef.nativeElement.querySelector('#form')
```
Comment (AlKulinski) [inline src/app/admin-api/admin-api.component.ts]: `const elementToSelect = this._elementRef.nativeElement.querySelector('#form')` 
What is the better way to access this element? I'd rather have it somewhere at the top of the file, like:
`@SomeDecorator(id) elementToSelect: HTMLElement`

**Review by EugeneHerasymchuk — APPROVED** [review]

File: src/app/directives/track-select.directive.ts [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  ngAfterViewInit (): void {
+    this.insertElement = this.el.nativeElement.querySelector(this.displayId)
+    this.focusWithin.trackSelect(this.el.nativeElement).subscribe((result: Element | null) => {
+      this.selectedElement = result || null
+      if (this.insertElement) { this.insertElement.innerHTML = result?.id ? result?.id : this.onEmpty }
+    })
+  }
+
+  // Simpler, but i wanted to try both to see if it works as expected
```
Comment (EugeneHerasymchuk) [inline src/app/directives/track-select.directive.ts:25]: 👍

File: src/app/directives/track-select.directive.ts [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+export class TrackSelectDirective implements AfterViewInit {
+  selectedElement: Element | null = null;
+  insertElement: Element | null = null;
+  @Input('displayId') displayId: string = '#insert';
+  @Input('onEmpty') onEmpty: string = 'nothing is selected' ;
+  constructor (private el: ElementRef, private focusWithin: FocusOnElementService) {
+
+  }
+
+  ngAfterViewInit (): void {
```
Comment (EugeneHerasymchuk) [inline src/app/directives/track-select.directive.ts:17]: just notice that lifecycle hooks are also in Directives, not only in Components

File: src/app/admin-api/admin-api.component.html [hunk 4]
```diff
@@ -1,6 +1,5 @@
-<h2>Wybrane: {{selectedName}}</h2>
-<form id="form" class="form" [formGroup]="formGroup" (ngSubmit)="onSubmit()">
-
+<form appTrackSelect displayId="#inputId" id="form" class="form" [formGroup]="formGroup" (ngSubmit)="onSubmit()">
```
Comment (EugeneHerasymchuk) [inline src/app/admin-api/admin-api.component.html:1]: 👍

**Review by EugeneHerasymchuk — APPROVED** [review]

**Pull request merged**